src/Chainweb/BlockHeader.hs

```diff
@@ -648,28 +648,29 @@ decodeBlockHeaderWithoutHash = do
         :+: a8
         :+: a9
         :+: a11
-        :+: a12
+        :+: a0
         :+: MerkleLogBody (blockHashRecordToVector a3)
 
 -- | Decode a BlockHeader and trust the result
 --
 decodeBlockHeader
     :: MonadGet m
     => m BlockHeader
-decodeBlockHeader = BlockHeader
-    <$> decodeNonce
-    <*> decodeBlockCreationTime
-    <*> decodeBlockHash -- parent hash
-    <*> decodeBlockHashRecord
-    <*> decodeHashTarget
-    <*> decodeBlockPayloadHash
-    <*> decodeChainId
-    <*> decodeBlockWeight
-    <*> decodeBlockHeight
-    <*> decodeChainwebVersion
-    <*> decodeEpochStartTime
-    <*> decodeFeatureFlags
-    <*> decodeBlockHash
+decodeBlockHeader = do
+    ff <- decodeFeatureFlags
+    ct <- decodeBlockCreationTime
+    ph <- decodeBlockHash -- parent hash
+    hr <- decodeBlockHashRecord
+    ht <- decodeHashTarget
+    pa <- decodeBlockPayloadHash
+    ci <- decodeChainId
+    wt <- decodeBlockWeight
+    hg <- decodeBlockHeight
+    cv <- decodeChainwebVersion
+    es <- decodeEpochStartTime
+    no <- decodeNonce
+    bh <- decodeBlockHash
+    pure $ BlockHeader no ct ph hr ht pa ci wt hg cv es ff bh
```

Comment:
I think changing the order of the fields in `BlockHeader` wouldn't break anything and would avoid these changes.

Reply:
Ah, yes, perhaps that's better.